src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -147,7 +147,7 @@ protected function doSave(array $values, $lifetime)
     private function getFile($id, $mkdir = false)
     {
         $hash = str_replace('/', '-', base64_encode(md5($id, true)));
```

Comment:
@nicolas-grekas converting the base64 to uppercase means that you have much more collision on the cache key (it is not about 2 bytes but about the whole path)